commit bcb37d7e236fd7811fc71279d87a043207f9d48d (refs/changes/35/1435/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-02-04T05:24:26+00:00 (2 years, 8 months ago)
    
    joyent/node-cueball#86 pool monitor broken on csets
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/CHANGES.adoc b/CHANGES.adoc
index fd8a7f8..796f9f1 100644
--- a/CHANGES.adoc
+++ b/CHANGES.adoc
@@ -6,6 +6,12 @@ toc::[]
 
 ## v2.x
 
+### v2.0.1
+
+Bugs fixed:
+
+ - #86 pool monitor broken on csets (kang endpoint always returns InternalError)
+
 ### v2.0.0
 
 New major release, due to API change in `ConnectionSet`.
diff --git a/lib/pool-monitor.js b/lib/pool-monitor.js
index 12baf97..51718fe 100644
--- a/lib/pool-monitor.js
+++ b/lib/pool-monitor.js
@@ -125,6 +125,8 @@ CueBallPoolMonitor.prototype.toKangOptions = function () {
 		});
 		ks.forEach(function (k) {
 			var fsm = cset.cs_fsm[k];
+			if (fsm === undefined)
+				return;
 			obj.fsms[k] = {};
 			var s = fsm.getState();
 			if (obj.fsms[k][s] === undefined)
diff --git a/package.json b/package.json
index c27d844..60352ae 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "cueball",
-  "version": "2.0.0",
+  "version": "2.0.1",
   "description": "",
   "main": "lib/index.js",
   "dependencies": {
